src/nodes.coffee

```diff
@@ -23,6 +23,10 @@ NO      = -> no
 THIS    = -> this
 NEGATE  = -> @negated = not @negated; this
 
+# Track comments that have been compiled into fragments, to avoid outputting
+# them twice.
+compiledComments = []
```

Comment:
I think this variable is local to this file, not global. We need to track what comments have been output because they get matched with tokens based on location data, and many tokens share the same location data hashes; so the same comment can appear on several tokens.